### DISCUSSION TITLE
Sensor Start/Stop events when using AAPS

### DISCUSSION THREAD

**Discussion by paravoid** [NightscoutFoundation/xDrip#2506] [Q&A]
**Background**: [AAPS' documentation](https://androidaps.readthedocs.io/en/latest/Configuration/xdrip.html) dictates to configure xDrip to not send treatments to Nightscout: _"Tap Extra Options -> Deactivate `Upload treatments` and `Back-fill data`."_. It says so with a Safety warning no less (_"Safety warning : You must deactivate “Upload treatments” from xDrip+, otherwise treatments can be doubled in AAPS leading to false COB and IOB."_).

This results in:
- xDrip+ sending SGV to Nightscout [_entries_]
- (optionally) xDrip+ sending the bridge's battery to Nightscout (through a separate setting, "Upload bridge battery") [_devicestatus_]
- AAPS sending all therapy events/treatments (boluses, temp basals, carb corrections, cannula changes, profile switches, notes, etc.) to Nightscout [_treatments_]
- AAPS sending the OpenAPS status [_devicestatus_]

Did I get everything right so far?

**First off somewhat of a basic question**: Is disabling "Upload treatments" when AAPS is used what the xDrip project recommends as well? I see in the code that xDrip+ has _some_ guards to avoid duplicate events, but it does feel like a fragile and potentially dangerous feature.

**Now onto the issue at hand**: I see xDrip+ has support for sending `Sensor Start` and `Sensor Stop` events, which in turn can be displayed by Nightscout using the SAGE plugin. This sounds useful! However, with the setup above, this doesn't work, as this is guarded by the `Upload treatments` setting, and thus these events aren't getting emitted by xDrip.

Is this the intended behavior? If not, may I suggest that perhaps `Sensor Start`/`Sensor Stop` events get behind a different setting (e.g. `Upload Sensor change events`) akin to the `Upload bridge battery` setting?

**Comment by Navid200** [comment]
@psonnera  Are the AAPS settings described correct?  Is anything overlooked?
@jwoglom Any opinions about the sensor start/stop settings?
@jamorham Do we need to do anything differently?  I mean there are so many other apps?  What are we going to do if every single one of them starts sending requests to xDrip to modify so that their app can function?  There is something inherently flawed about that thought.  Why does AAPS need xDrip to change continuously?  Do we ever ask AAPS to change?

**Reply by psonnera** [reply]
With AAPS xDrip+ should only be used to upload BG to Nightscout. Think of it as the Dexcom share plugin: CGM only. Treatments are handled only by AAPS. Sensor change are handled by Nightscout.
@Navid200 there's no reason xDrip+ should be customized to adapt to AAPS but maybe having a selection of which events/treatments should be sent to Nightscout wouldn't be a bad thing. xDrip+ is born to handle sensor events. BG mainly. Start and stop are sensor events and could be sent by default to Nightscout unlike other treatments (carbs, UI).
Just a thought.

**Reply by paravoid** [reply]
@psonnera that matches my understanding. Having a selection of events/treatments makes sense to me. Sensor start/stop events are what I consider part of CGM, and information for which xDrip is authoritative (as opposed to e.g. boluses, for which AAPS is the primary source of truth, with xDrip receiving it from AAPS). Hence my surprise!

**Comment by Navid200** [comment]
@psonnera xDrip is considered to be extremely complicated by the  masses.
There are xDrip forum mods who recommend build your own Dexcom app to people.  Considering they are mods, there is something telling in that.
The regular people say I need something that shows my blood glucose and has alerts and just works.

The problem we have is that we have allowed "just one more setting" to get us here.
We can keep adding settings.  But, I don't recommend it.  You know that there is a resistance in retiring old settings.

What is the point of developing an extremely sophisticated plane that the majority have no interest in boarding because they prefer something that just gets them to where they need to go with no bells and whistles that they never need to use?
  
Please support me in asking senior members to limit new settings to those that are absolutely necessary.

**Reply by psonnera** [reply]
Understood, agreed. Then why not evaluate sending sensor start/stop events to Nightscout as a default behavior?

Still, agreed. Too many settings are only making xDrip+ too complicated. Point taken.
